src/lib/dnssd/platform/Dnssd.h

```diff
@@ -102,10 +102,18 @@ using DnssdResolveCallback = void (*)(void * context, DnssdService * result, con
  * The callback function SHALL NOT take the ownership of the service pointer or
  * any pointer inside this structure.
  *
+ * It is possible that this callback will be called multiple times. In case when
```

Comment:
I find this awkward/problematic to maintain. I expect errors to actually mean errors. This seems to be some form of ternary of in one of `No error, more callbacks expected`, `no error, final callback`, `has error, final callback`.

I would prefer some better split of arguments and their meaning. Having `CHIP_END_OF_INPUT` be used as an error code that means success is not ideal.

Reply:
I've added new parameter for the callback function, so right now the "end-of-input" will be indicated in an explicit way.